lib/hal_api/controller/exceptions.rb

```diff
@@ -42,6 +44,12 @@ def log_error(env, wrapper)
     end
   end
 
+  def notice_error(error)
+    ::NewRelic::Agent.notice_error(error)
```

Comment:
I think you could save yourself a raise and rescue with:
```ruby
::NewRelic::Agent.notice_error(error) if defined?(::NewRelic::Agent)
```
or get a bit fancier
```ruby
if defined?(::NewRelic::Agent) && ::NewRelic::Agent.respond_to?(:notice_error)
  ::NewRelic::Agent.notice_error(error)
end
```